modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ModelBuilderContext/objectFolderReducer.ts

```diff
@@ -122,18 +125,39 @@ export function ObjectFolderReducer(state: TState, action: TAction): TState {
 					} as ObjectFieldNodeRow;
 				}
 			);
-			const updatedObjectDefinitionsNodes = elements.map((node) => {
+
+			selectedObjectFolder.objectFolderItems.push({
+				linkedObjectDefinition:
+					selectedObjectFolder.externalReferenceCode !==
+					newObjectDefinition.objectFolderExternalReferenceCode,
+				objectDefinitionExternalReferenceCode:
+					newObjectDefinition.externalReferenceCode,
+				positionX: newPosition.x,
+				positionY: newPosition.y,
+			});
+
+			const updatedObjectFolders = objectFolders.filter(
+				(objectFolder) => {
+					objectFolder.externalReferenceCode !==
+						selectedObjectFolder.externalReferenceCode;
+				}
+			);
+
+			updatedObjectFolders.push({
+				...selectedObjectFolder,
+				objectFolderItems: selectedObjectFolder.objectFolderItems,
```

Comment:
is this line really necessary? since you are already spreading the `selectedObjectFolder` and inside it has `objectFolderItems`? maybe i'm understanding wrong, but i think you could just pass the  `selectedObjectFolder` on the push like this:

`updatedObjectFolders.push(selectedObjectFolder);`

Reply:
Hey @rebsilva, you are right!